packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useMemo} from 'react'
+import logger from '@salesforce/retail-react-app/app/utils/logger-instance'
+import {initDataCloudSdk} from '@salesforce/cc-datacloud-typescript'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+import {useUsid, useCustomerId, useCustomerType, useDNT} from '@salesforce/commerce-sdk-react'
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
+
+export class DataCloudApi {
+    constructor({site}) {
+        this.site = site
+
+        // Grab Data Cloud configuration values and intialize the sdk
+        const {
+            app: {dataCloudAPI: config}
+        } = getConfig()
```

Comment:
Do you think it would be better to pass the configuration as arguments instead of relying on using the getConfig internally? It will most likely make testing easier.